src/main/java/com/ait/lienzo/client/widget/panel/mediators/PanelMediators.java

```diff
@@ -0,0 +1,176 @@
+import com.google.gwt.event.dom.client.MouseOutHandler;
+import com.google.gwt.event.shared.HandlerRegistration;
+
+public class PanelMediators {
+
+    static final double MIN_SCALE = 0.1;
+    static final double MAX_SCALE = Double.MAX_VALUE;
+    static final double ZOOM_FACTOR = 0.1;
+
+    private Supplier<LienzoBoundsPanel> panelSupplier;
+    private MouseWheelZoomMediator zoomMediator;
+    private MousePanMediator panMediator;
+    private PanelPreviewMediator previewMediator;
+    private HandlerRegistration outHandler;
+
+    public static PanelMediators build(final LienzoBoundsPanel panel) {
+        return new PanelMediators().init(new Supplier<LienzoBoundsPanel>() {
+            @Override
+            public LienzoBoundsPanel get() {
+                return panel;
+            }
+        });
+    }
+
+    public PanelMediators init(final Supplier<LienzoBoundsPanel> panelSupplier) {
+        return init(panelSupplier,
+                    new Supplier<PanelPreviewMediator>() {
+                        @Override
+                        public PanelPreviewMediator get() {
+                            return PanelPreviewMediator.build((ScrollablePanel) panelSupplier.get());
+                        }
+                    });
+    }
+
+    public PanelMediators init(final Supplier<LienzoBoundsPanel> panelSupplier,
+                               final Supplier<PanelPreviewMediator> previewMediatorBuilder) {
+        this.panelSupplier = panelSupplier;
+        final LienzoBoundsPanel panel = panelSupplier.get();
+
+        final Viewport viewport = getViewport();
+
+        zoomMediator = new MouseWheelZoomMediator()
+                .setMinScale(MIN_SCALE)
+                .setMaxScale(MAX_SCALE)
+                .setZoomFactor(ZOOM_FACTOR);
+
+        panMediator = new MousePanMediator()
+                .setXConstrained(true)
+                .setYConstrained(true);
+
+        zoomMediator.setEnabled(false);
+        panMediator.setEnabled(false);
+
+        viewport.getMediators().push(zoomMediator);
+        viewport.getMediators().push(panMediator);
+
+        if (panel instanceof ScrollablePanel) {
+            previewMediator = previewMediatorBuilder.get();
+        }
+
+        outHandler = panel.addMouseOutHandler(new MouseOutHandler() {
+            @Override
+            public void onMouseOut(MouseOutEvent mouseOutEvent) {
+                disable();
+            }
+        });
+        return this;
+    }
+
+    public boolean enableZoomMediator() {
+        if (panMediator.isEnabled() || previewMediator.isEnabled()) {
+            return false;
+        }
+        zoomMediator.setEnabled(true);
+        return true;
+    }
+
+    public boolean enablePanMediator() {
+        if (zoomMediator.isEnabled() || previewMediator.isEnabled()) {
+            return false;
+        }
+        panMediator.setEnabled(true);
+        return true;
+    }
+
+    public boolean enablePreviewMediator() {
```

Comment:
Return value of the method is never used

Reply:
it's used on Stunner side